app/jobs/ingest_job.rb

```diff
@@ -0,0 +1,17 @@
+class IngestJob < Hyrax::ApplicationJob
+  queue_as Hyrax.config.ingest_queue_name
+
+  after_perform do |job|
+    opts = job.arguments.reverse.detect { |x| x.is_a? Hash } || {}
+    wrapper = job.arguments.first
+    ContentNewVersionEventJob.perform_later(wrapper.file_set, wrapper.user) if opts[:notification]
```

Comment:
@atz Right. At the moment, there are six event jobs around content changes:

https://github.com/samvera/hyrax/blob/master/config/initializers/hyrax_callbacks.rb#L2-L24

* `:after_create_concern` creates an event for every new work, and it fires from the actor stack: https://github.com/samvera/hyrax/blob/68527862df498aa9f407b909d4f41d781ba146dc/app/actors/hyrax/actors/base_actor.rb#L14
* `:after_create_fileset` creates an event when a fileset is attached to a work using `attach_to_work`: https://github.com/samvera/hyrax/blob/77140b38907158aaa4e30aae442ea5f04d0b1be2/app/actors/hyrax/actors/file_set_actor.rb#L58
* `:after_revert_content` creates an event when a fileset is reverted to a past version: https://github.com/samvera/hyrax/blob/77140b38907158aaa4e30aae442ea5f04d0b1be2/app/actors/hyrax/actors/file_set_actor.rb#L69
* `:after_update_content` you're well acquainted with
* `:after_update_metadata` creates an event when a work is updated (as distinct from when a fileset is updated), invoked by the actor stack: https://github.com/samvera/hyrax/blob/68527862df498aa9f407b909d4f41d781ba146dc/app/actors/hyrax/actors/base_actor.rb#L22
* `:after_destroy` creates an event when a work is destroyed, called from the works controller: https://github.com/samvera/hyrax/blob/45b988145d76dd5c2e659aa0cd53737e7d138c3a/app/controllers/concerns/hyrax/works_controller_behavior.rb#L117

Part of why we're using the callbacks in different spots in the codebase could be that some of our actors are meant to be used in a stack and others are meant to be used alone (more akin to "service objects"?). I'm definitely not suggesting this PR needs to tackle this problem. ;) I'm leaving this comment here to show that we considered changes to how we approach this but that we decided to punt for good reason.